resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -367,7 +369,7 @@ public Match match(List<Match> matches, String httpMethod, HttpRequest request)
       for (MediaType accept : requestAccepts) weightedAccepts.add(WeightedMediaType.parse(accept));
 
       List<Match> list = new ArrayList<Match>();
-
+      Set<MediaType> patchAccepts = new HashSet<MediaType>();

```

Comment:
The HashSet here is backed by a HashMap of size 16. Maybe we can leave 'patchAccepts' as null here and assign it to an actual HashSet instance only when we really need it later (at that time we also know the initial size for the map btw) ?

Reply:
OK. I changed it to create patchAccepts hash set with initialize size 8 only if the http method is "option" and "patch" is listed in the allowed method list.